pkg/node/manager/manager.go

```diff
@@ -281,6 +281,11 @@ func (m *Manager) NodeUpdated(n node.Node) {
 	var nodeIP, nodeIP4 net.IP
 	dpUpdate := true
 
+	remoteHostIdentity := identity.ReservedIdentityHost
+	if option.Config.EnableNonLocalNodeIdentity {
+		remoteHostIdentity = identity.ReservedIdentityNonLocalNode
+	}
```

Comment:
Should we also start to treat `pkg/node/addressing.NodeInternalIP` as `identity.ReservedIdentityNonLocalNode`? the code just below entirely skips propagating IPcache entries for these IP types, which includes network-facing IPs in ENI environments (amongst other cases).

Reply:
Yes. I see that the internal IP is only added when encryption is enabled which seems wrong.